src/client/datascience/interactive-window/interactiveWindow.ts

```diff
@@ -94,10 +84,9 @@ export class InteractiveWindow implements IInteractiveWindowLoadable {
         return this._onDidChangeViewState.event;
     }
     // Promise that resolves when the interactive window is ready to handle code execution.
-    public get readyPromise(): Promise<NotebookEditor> {
-        return this._editorReadyPromise;
+    public get readyPromise(): Promise<void> {
+        return Promise.all([this._editorReadyPromise, this._kernelReadyPromise]).then(noop, noop);
```

Comment:
Previously the kernel promise was part of the editor ready promise, so if the kernel failed to start e.g. due to ipykernel not being installed, subsequent executions would fail while awaiting on the editor ready (if the kernel promise successfully resolved, there wasn't an issue). PR splits out the kernel promise